### PR TITLE
fixes: page_id and page

### PR DESCRIPTION
**Pull request by gabriele-0201**

### CODE REVIEW

**Comment by gabriele-0201** [comment]
* **#35** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/35?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#34** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/34?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @gabriele-0201 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: core/src/page_id.rs [hunk 1]
```diff
@@ -58,28 +86,15 @@ pub fn parent_page_id(mut page_id: PageId) -> PageId {
...
+/// Iterator of PageIds over a KeyPath,
+/// PageIds will be lazily constructed as needed
+pub struct PageIdsIterator<'a> {
     sextets: ChunksExact<'a, u8, Msb0>,
     page_id: Option<PageId>,
 }
 
 impl<'a> PageIdsIterator<'a> {
-    fn new(key_path: &'a KeyPath) -> Self {
+    pub fn new(key_path: &'a KeyPath) -> Self {
```
Comment (rphmeier) [inline core/src/page_id.rs:97]: docs on public items

Comment (gabriele-0201) [reply core/src/page_id.rs:97]: miss it 😅 
pushed comments!

**Review by rphmeier — APPROVED** [review]
Thanks!

**Pull request merged**